content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx

```diff
@@ -39,13 +24,47 @@ model User {
+
+- set constraints that prevent you from making changes that break references
+- set [referential actions](/concepts/components/prisma-schema/relations/referential-actions) that define how changes to records are handled
+
+Together these constraints and referential actions guarantee the _referential integrity_ of the data.
+
+For the example schema above, Prisma Migrate will generate the following SQL by default if you use the PostgreSQL connector:
 
-When using Prisma Client the behavior of queries is identical or similar, but all referential actions and most referential integrity is actually handled by the Prisma engine under the hood.
+```sql highlight=19;normal
+
+-- CreateTable
+CREATE TABLE "Post" (
+    "id" SERIAL NOT NULL,
+    "title" TEXT NOT NULL,
+    "authorId" INTEGER NOT NULL,
+
+    CONSTRAINT "Post_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "User" (
+    "id" SERIAL NOT NULL,
+
+    CONSTRAINT "User_pkey" PRIMARY KEY ("id")
+);
+
+-- AddForeignKey
+ALTER TABLE "Post" ADD CONSTRAINT "Post_authorId_fkey" FOREIGN KEY ("authorId") REFERENCES "User"("id") ON DELETE CASCADE ON UPDATE CASCADE;
+
+```
+
+In this case, the foreign key constraint on the `authorId` column of the `Post` table references the `id` column of the `User` table, and guarantees that a post must have an author that exists. If you update or delete a user then the `ON DELETE` and `ON UPDATE` referential actions specify the `CASCADE` option, which will also delete or update all posts belonging to the user.
+
+Some databases, such as MongoDB or [PlanetScale](/guides/database/using-prisma-with-planetscale#differences-to-consider)),
+do not support foreign keys. Additionally, in some cases developers may prefer not to use foreign keys in their relational database. For these situations, Prisma offers [an alternative relation mode, `prisma`](#emulate-relations-in-prisma-with-the-prisma-relation-mode), which emulates most properties of relations in relational databases. When you use the Prisma Client with the `prisma` relation mode enabled, the behavior of queries is identical or similar, but referential actions and most constraints are handled by the Prisma engine rather than in the database.
 
 <Admonition type="warning">
-  There are performance implications to emulating referential integrity and referential actions in the Prisma client. In cases where the underlying database supports foreign keys, it is usually the preferred choice.
+  There are performance implications to emulating referential integrity and
+  referential actions in the Prisma Client. In cases where the underlying
+  database supports foreign keys, it is usually the preferred choice.
```

Comment:
Weird that we are formatting text into multiple lines in Admonitions, but not anywhere else (which I _very_ much prefer).

Reply:
Formatter just does that, don't know why